diff -ur squidGuard-1.5-beta.orig/src/sgDb.c squidGuard-1.5-beta/src/sgDb.c
--- squidGuard-1.5-beta.orig/src/sgDb.c      2017-02-01 07:17:51.800736824 +0100
+++ squidGuard-1.5-beta/src/sgDb.c      2017-02-01 07:18:31.300737356 +0100
@@ -112,7 +112,7 @@
     }
   }
 #endif
-#if DB_VERSION_MAJOR == 4
+#if DB_VERSION_MAJOR >= 4
   if(globalUpdate || createdb || (dbfile != NULL && stat(dbfile,&st))){
     flag = DB_CREATE;
     if(createdb)

