diff --git a/setup.py b/setup.py
index 661e5b6..4e60f1e 100644
--- a/setup.py
+++ b/setup.py
@@ -50,7 +50,7 @@ metadata = {
         "Programming Language :: Python :: Implementation :: PyPy",
     ],
     "keywords": "http api rest client retrofit",
-    "packages": find_packages(exclude=("tests",)),
+    "packages": find_packages(exclude=("tests*",)),
     "install_requires": install_requires,
     "extras_require": extras_require,
 }
