--- a/setup.py	2024-08-18 23:19:55.000000000 +0200
+++ a/setup.py.new	2024-09-11 20:39:36.755962293 +0200
@@ -21,7 +21,7 @@
     description="Asynchronous library to control Shelly devices.",
     long_description=README_FILE.read_text(encoding="utf-8"),
     long_description_content_type="text/markdown",
-    packages=find_packages(),
+    packages=find_packages(exclude=["tests", "tests.*"]),
     python_requires=">=3.11",
     package_data={"aioshelly": ["py.typed"]},
     zip_safe=True,
